### ISSUE TITLE
Duplicate display of the personalization and does not save the personalizations when several combinations

### ISSUE THREAD

**Issue by Anthony-Tran** [PrestaShop/PrestaShop#30028]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

2 bugs identified.
The first displays duplicate customization on cart in front office. When the order is made, in the back office, no problem because the customizations are well separated. So only a display problem

The second bug is much more problematic. It does not display all the customizations in the cart and when ordering, the registration is not complete.

I came across a thread that looks like this : https://github.com/PrestaShop/PrestaShop/issues/28323

### Steps to reproduce

Version 1.7.8.7, clean installation. 

First bug (display)
============
Product with a personalization text field, 2 combinations (color and size for example).

- On the front office. Save personalization text. Choose a size and color and then add to cart.
- Do it a second time by putting another text and choosing a different size. Add to Cart.
Look at your basket and you will see the customization displayed twice on the 2 products

I made video to show you : https://www.prestashop.com/forums/uploads/monthly_2022_10/v-1-7-8-7-double.mp4.6ff9fb24e91ef32eed056e0f19243727.mp4

________________________
Second bug
========
Product with a personalization text field, 3 combinations (color, size and dimension for example).
- On the front office. Save personalization text. Choose a size and color and dimension then add to cart.
- Do it a second time by putting another text, choosing a different size, different color and different dimension. Add to Cart.
Look at your basket and there will only be a personalization.
If the order is made, only one personalization text will be saved.

I made video to show you : https://www.prestashop.com/forums/uploads/monthly_2022_10/erase-custom.mp4.8f8ca711bf1671e20fc8a68d5a749d1d.mp4

________________
Full video pour the second bug with back office : https://www.prestashop.com/forums/uploads/monthly_2022_10/full.mp4.b6a0b851862d231373e232723cfc6a9a.mp4

### PrestaShop version(s) where the bug happened

1.7.8.7

### PHP version(s) where the bug happened

7.1

**Comment by AureRita** [comment]
Hi @Anthony-Tran 

Your issue seems to be similar than this one : #29185 

Can you say me if it's the same or not ? 

Waiting for your feedback

**Comment by Anthony-Tran** [comment]
Hi @AureRita 
Part of the problem is similar (double customization).
The 2nd bug is not addressed in this post and it is the one that is the most problematic because it does not save customization.

**Comment by hibatallahAouadni** [comment]
Hello @Anthony-Tran 

About the first bug is a duplicate of this one #29185 and it was fixed by this PR #29635
The second bug, I reproduce the issue only with PS1787 and PS800. I’ll add this to the debug backlog so that it’s fixed.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. 

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

PS: Related to #28688

Thank you

**Comment by Anthony-Tran** [comment]
After several hours of understanding how compilation works, etc... I managed to fix the bug by dissecting the various tickets.
So I'll write the solution in case someone "newbie" like me can rectify this problem.
In ticket [#29635](https://github.com/PrestaShop/PrestaShop/pull/29635) the issue is resolved. [3 files](https://github.com/PrestaShop/PrestaShop/pull/29635/files) are to be modified. ProductController.php, GetCartForViewingHandler.php
and product.js.
The first 2 can be modified directly. On the other hand the last one requires a compilation. I made the compilation so I will give you directly the file to modify to have this modification compiled in your prestashop.
File: prestashop/themes/core.js
```js
// Replace this line:
(0,r.default)(o.default.selectors.product.inputCustomization).val(0),

// By this line:
"updatedProductQuantity"!==t&&"updatedProductCombination"!==t||!e.id_customization?(0,r.default)(o.default.selectors.product.inputCustomization).val(0):(0,r.default)(o.default.selectors.cart.productCustomizationId).val(e.id_customization),
```

**Comment by MatShir** [comment]
Hi @Anthony-Tran, 
You could try to make a PR, we will be happy to review your work ;)

**Comment by littlebull** [comment]
Hi all,

we experience the same problem as mentioned by others.

But if I am understanding the code correctly, the problem will also be the index id_cart_product in the table ps_customization. The index is on the columns id_cart, id_product, id_product_attribute. So the DB prohibits more than one customization on the same product in one cart. Am I right?

Julian

**Comment by Dreimus** [comment]
Hello, 

Any news regarding this issue ?

**Comment by Hlavtox** [comment]
@PrestaShop/qa-functional Guys can you retest on 8.1x and develop? There were some big changes to customization logic. Thanks!

**Comment by florine2623** [comment]
Hello all,

**BUG 1 :** ✅ 

Using PS `1.7.8.9`, I can't reproduce the first issue. The customization is not duplicated.

https://github.com/PrestaShop/PrestaShop/assets/16019289/f124f1cd-ce76-4998-8838-d8ea5a620de8

Using `8.1.x` and `develop` with PPV2, the customization field in cart is not duplicated either.
<img width="1113" alt="Screenshot 2023-07-12 at 11 10 23" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/0d8445f3-6d43-4c90-a004-ceca6e2491e1">

**BUG 2 :** ✅ 

Using PS `1.7.8.9`, I can't reproduce the first issue. Both custom fields are well saved on cart and during checkout.

FO, catalog :

https://github.com/PrestaShop/PrestaShop/assets/16019289/5200bcbd-600c-4f5b-951b-e7c33b3a42b8

FO, checkout process :

https://github.com/PrestaShop/PrestaShop/assets/16019289/d0bffc8b-0967-4f4a-97cc-6ba396e23e3c

Using `8.1.x` and `develop` branch, I do not have the issue either. It works as expected.

https://github.com/PrestaShop/PrestaShop/assets/16019289/6cf2e263-d8d0-42cb-afef-91c71f73b493

**Comment by MatShir** [comment]
It seems to be fixed on 8.1 🥳 
We can close the issue :)
